node/src/components/contract_runtime/operations.rs

```diff
@@ -81,6 +81,12 @@ pub fn execute_finalized_block(
     let native_runtime_config = NativeRuntimeConfig::from_chainspec(chainspec);
     let addressable_entity_enabled = chainspec.core_config.enable_addressable_entity();
 
+    if addressable_entity_enabled != data_access_layer.enable_addressable_entity {
+        return Err(BlockExecutionError::InvalidAESetting(
+            data_access_layer.enable_addressable_entity,
+        ));
+    }
```

Comment:
It seems odd to me that this would even be possible. Would a debug assert be more appropriate here?

Reply:
This was my change, I can amend it to be a debug assert, but just preventing execution of any transaction made sense at the time

Reply:
I wanted this here primarily for programmer error, just incase a later miswire was introduced, but its not a show stopper for me

Reply:
lets leave it for now...but i would like to revisit

Reply:
Understood